jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java

```diff
@@ -104,20 +117,24 @@ public static Booster train(
     }
 
     //initialize booster
-    Booster booster = new Booster(params, allMats);
-
-    int version = booster.loadRabitCheckpoint();
+    if (booster == null) {
+      booster = new Booster(params, allMats);
+      booster.loadRabitCheckpoint();
+    } else {
```

Comment:
add comments to specify this is the branch for training based on an existing model

Reply:
Done.